op-chain-ops/upgrades/l1.go

```diff
@@ -302,11 +348,60 @@ func SystemConfig(batch *safe.Batch, implementations superchain.ImplementationLi
 		return fmt.Errorf("no initialize method")
 	}
 
-	gasPriceOracleOverhead := new(big.Int).SetUint64(config.GasPriceOracleOverhead)
-	gasPriceOracleScalar := new(big.Int).SetUint64(config.GasPriceOracleScalar)
-	batcherHash := common.BytesToHash(config.BatchSenderAddress.Bytes())
-	l2GenesisBlockGasLimit := uint64(config.L2GenesisBlockGasLimit)
-	startBlock := new(big.Int).SetUint64(config.SystemConfigStartBlock)
+	var gasPriceOracleOverhead, gasPriceOracleScalar, startBlock *big.Int
+	var batcherHash common.Hash
+	var batchInboxAddress, p2pSequencerAddress, finalSystemOwner common.Address
+	var l2GenesisBlockGasLimit uint64
+
+	if config != nil {
+		gasPriceOracleOverhead = new(big.Int).SetUint64(config.GasPriceOracleOverhead)
+		gasPriceOracleScalar = new(big.Int).SetUint64(config.GasPriceOracleScalar)
+		batcherHash = common.BytesToHash(config.BatchSenderAddress.Bytes())
+		l2GenesisBlockGasLimit = uint64(config.L2GenesisBlockGasLimit)
+		startBlock = new(big.Int).SetUint64(config.SystemConfigStartBlock)
+		batchInboxAddress = config.BatchInboxAddress
+		p2pSequencerAddress = config.P2PSequencerAddress
+		finalSystemOwner = config.FinalSystemOwner
+	} else {
+		systemConfig, err := bindings.NewSystemConfigCaller(common.HexToAddress(chainConfig.SystemConfigAddr.String()), backend)
+		if err != nil {
+			return err
+		}
+		gasPriceOracleOverhead, err = systemConfig.Overhead(&bind.CallOpts{})
+		if err != nil {
+			return err
+		}
+		gasPriceOracleScalar, err = systemConfig.Scalar(&bind.CallOpts{})
+		if err != nil {
+			return err
+		}
+		batcherHash, err = systemConfig.BatcherHash(&bind.CallOpts{})
+		if err != nil {
+			return err
+		}
+		l2GenesisBlockGasLimit, err = systemConfig.GasLimit(&bind.CallOpts{})
+		if err != nil {
+			return err
+		}
+		// StartBlock is a new property, we want to explicitly set it to 0 if there is an error fetching it
```

Comment:
Is there a way to check if it doesn't exist rather than if there was an RPC/other error?
Maybe we should be checking the semver of the systemConfig to figure out if it is a hard error or a temp error.

Reply:
I definitely see your point here, it becomes more explicit with the versions. I have a preference to not add branching logic based on versions for now, if it becomes an issue when running this in practice then i will do that